### ISSUE TITLE
Bundler: revisit why bundle console was deprecated

### ISSUE THREAD

**Issue by tnir** [rubygems/rubygems#5777]
Actually I was thinking about un-deprecating this command since I'm using it often. It suggest to be replaced by `bin/console` generated by the `bundle gem`, but that is not often available and `bundle console` is the only way to open console quickly.

Usually my usecase is to quickly prototype some code starting with `Gemfile` to fetch deps and opening the console via `bundle console`.

_Originally posted by @simi in https://github.com/rubygems/rubygems/issues/5775#issuecomment-1195617896_

**Comment by tnir** [comment]
> [...] `bin/console` [...] is not often available [...]

@simi What case do you mention? This mush be a kind of bikeshedding discussion 😁

**Comment by simi** [comment]
I'm not sure what's `bikeshedding` in here. `bin/console` is only available when created manually or by `bundle gem`. There are valid usecases (as the one I have mentioned) when `bin/console` is not present and `bundle console` is still elegant way to jump in into console with deps loaded.

**Comment by deivid-rodriguez** [comment]
I think bikeshedding in this context means that people would request many features to `bundle console` that can be easily added to a `bin/console` script, but impose a maintenance overhead if added to the builtin command. This is what I recall from when I wrote the UPGRADING.md document. When I wrote that document I went through all deprecations, read the relevant Github issues and tried to summarize what was decided there. But honestly I don't recall the specifics of `bundle console`.

I think we can look for the relevant issues/discussions again and decide if they still make sense these days.
